### DISCUSSION TITLE
Encrypted property prototype

### DISCUSSION THREAD

**Discussion by jasonwilson** [mikro-orm/mikro-orm#3985] [Ideas]
Hey @B4nan,

We've been using mikro-orm for a couple years and recently needed the ability for column level encryption. I created a little prototype that can be found here: https://gist.github.com/jasonwilson/e2f6fbb704beb00578d4bc6d903942e2

Does this seem like something you would potentially want to add to mikro-orm? If so, I could open a pull request or publish a third party package.

**Comment by B4nan** [comment]
Is this really working? :] Seems wrong to use the `after*` events, your changes would make it only to the next flush, not the current one that triggered those events.

I am not entirely sure if I like something like that inside the core, the idea with encryption support sounds good to me, but I feel like the solution would have to be much more generic. Not sure how other tools are approaching encryption, but I'd expect users to want to have control over how it's implemented. Imagine you have an existing database, you won't be up to rehashing everything...

**Reply by jasonwilson** [reply]
It does seem to work but I've updated my implementation to use the `before` lifecycle hooks. Good catch.
